--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1997,6 +1997,9 @@ ENDIF(WIN32 AND NOT CYGWIN)
 #
 #
 INCLUDE_DIRECTORIES(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/libarchive)
+if(NOT WIN32)
+  INCLUDE_DIRECTORIES(AFTER ${CMAKE_CURRENT_SOURCE_DIR}/contrib/android/include)
+ENDIF()
 #
 IF(MSVC)
   ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE)
--- a/libarchive/CMakeLists.txt
+++ b/libarchive/CMakeLists.txt
@@ -9,6 +9,7 @@
 SET(include_HEADERS
   archive.h
   archive_entry.h
+  ../contrib/android/include/android_lf.h
 )
 
 # Sources and private headers
