diff --git a/src/common/mythread.h b/src/common/mythread.h
index be22654..3c1cef8 100644
--- a/src/common/mythread.h
+++ b/src/common/mythread.h
@@ -332,8 +332,12 @@ mythread_condtime_set(mythread_condtime *condtime, const mythread_cond *cond,
 #define WIN32_LEAN_AND_MEAN
 #ifdef MYTHREAD_VISTA
 #	undef _WIN32_WINNT
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+#	define _WIN32_WINNT 0x0A00
+#else
 #	define _WIN32_WINNT 0x0600
 #endif
+#endif
 #include <windows.h>
 #include <process.h>
 
@@ -349,6 +353,11 @@ typedef HANDLE mythread_cond;
 typedef CONDITION_VARIABLE mythread_cond;
 #endif
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+#undef GetTickCount
+#define GetTickCount() ((DWORD)GetTickCount64())
+#endif
+
 typedef struct {
 	// Tick count (milliseconds) in the beginning of the timeout.
 	// NOTE: This is 32 bits so it wraps around after 49.7 days.
-- 
2.13.2.windows.1

