--- a/contrib/cmake_unofficial/CMakeLists.txt
+++ b/contrib/cmake_unofficial/CMakeLists.txt
@@ -123,6 +123,7 @@ else()
   set(LZ4_LINK_LIBRARY lz4_static)
 endif()
 
+if(ENABLE_EXECUTABLES)
 # lz4
 set(LZ4_PROGRAMS_BUILT lz4cli)
 add_executable(lz4cli ${LZ4_CLI_SOURCES})
@@ -136,6 +137,7 @@ if (LZ4_BUILD_LEGACY_LZ4C)
   set_target_properties(lz4c PROPERTIES COMPILE_DEFINITIONS "ENABLE_LZ4C_LEGACY_OPTIONS")
   target_link_libraries(lz4c ${LZ4_LINK_LIBRARY})
 endif()
+endif()
 
 # Extra warning flags
 include (CheckCCompilerFlag)
@@ -171,9 +173,11 @@ endforeach (flag)
 if(NOT LZ4_BUNDLED_MODE)
   include(GNUInstallDirs)
 
+  if(ENABLE_EXECUTABLES)
   install(TARGETS ${LZ4_PROGRAMS_BUILT}
     BUNDLE	DESTINATION "${CMAKE_INSTALL_BINDIR}"
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
+  endif()
   install(TARGETS ${LZ4_LIBRARIES_BUILT}
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
